alerting/src/test/kotlin/org/opensearch/alerting/util/clustermetricscoordinatortest/ClusterMetricsCoordinatorIT.kt

```diff
@@ -0,0 +1,209 @@
+            val source = doc["_source"] as Map<String, Map<String, Any>>
+            val metricType = source.keys.first()
+            if (metricType == "cluster_status") {
+                if (source[metricType]?.get("timestamp") == time) {
+                    assertTrue(flag)
+                    return
+                }
+            }
+        }
+        flag = true
+        assertTrue(flag)
+    }
+
+    fun `test frequency`() {
+        client().updateSettings("plugins.alerting.cluster_metrics.execution_frequency", "2m")
+        Thread.sleep(300000)
+
+        val response = getResponse()
+        val hits = getHits(response)
+        val docs = hits["hits"] as ArrayList<Map<String, Any>>
+        val times = mutableSetOf<String>()
+
+        for (doc in docs) {
+            val source = doc["_source"] as Map<String, Map<String, Any>>
+            val metricType = source.keys.first()
+            times.add(source[metricType]?.get("timestamp").toString())
+        }
+        logger.info("this is times $times")
+        val time1 = Instant.parse(times.elementAt(times.size - 1))
+        val time2 = Instant.parse(times.elementAt(times.size - 2))
+        val diff = time2.until(time1, ChronoUnit.MINUTES)
+        assertEquals(diff, 2)
+    }
+
+    fun `test update storage time to less than minimum storage time`() {
+        assertThrows(IllegalArgumentException::class.java) {
+            try {
+                client().updateSettings("plugins.alerting.cluster_metrics.metrics_history_max_age", "30s")
+            } catch (t: ResponseException) {
+                logger.info("LOG THE RESPONSE EXCEPTION OBJECT ${t.cause}")
+            }
+        }
+    }
+
+    fun `test update frequency to less than minimum frequency`() {
+        assertThrows(IllegalArgumentException::class.java) {
+            client().updateSettings("plugins.alerting.cluster_metrics.execution_frequency", "1ms")
+        }
+    }
+
+    fun `test when execution frequency greater than storage time`() {
+        assertThrows(IllegalArgumentException::class.java) {
+            client().updateSettings("plugins.alerting.cluster_metrics.metrics_history_max_age", "20m")
+            client().updateSettings("plugins.alerting.cluster_metrics.execution_frequency", "25m")
+        }
+    }
+
+    fun `test successful client setting update`() {
+        client().updateSettings("plugins.alerting.cluster_metrics.metrics_history_max_age", "400m")
+        client().updateSettings("plugins.alerting.cluster_metrics.execution_frequency", "12m")
+        val response = getSettings()
+        val persistentMap = response["persistent"] as Map<String, Any>
+        val executionFrequency = persistentMap["plugins.alerting.cluster_metrics.execution_frequency"].toString()
+        val storageTime = persistentMap["plugins.alerting.cluster_metrics.metrics_history_max_age"].toString()
+        assertEquals(executionFrequency, "12m")
+        assertEquals(storageTime, "400m")
+    }
+
```

Comment:
When manually testing, you used the following command to update the `execution_frequency` and `metrics_history_max_age` settings at the same time. Could you implement a test that confirms they can't be set to invalid values when updating both settings at once? You'd essentially want to ensure the following command fails as expected:
```
PUT _cluster/settings
{
  "persistent": {
    "plugins.alerting.cluster_metrics.execution_frequency": "10m",
    "plugins.alerting.cluster_metrics.metrics_history_max_age": "5m"
  }
}

Reply:
Done in latest commit.